app/src/org/commcare/graph/model/GraphData.java

```diff
@@ -63,4 +73,91 @@ public String getConfiguration(String key, String defaultValue) {
+        SortedMap<String, String> variables = new TreeMap<>();
+        JSONObject config = new JSONObject();
+        StringBuilder html = new StringBuilder();
+        try {
+            // Configure data first, as it may affect the other configurations
+            DataConfiguration data = new DataConfiguration(this);
+            config.put("data", data.getConfiguration());
+
+            AxisConfiguration axis = new AxisConfiguration(this);
+            config.put("axis", axis.getConfiguration());
+
+            GridConfiguration grid = new GridConfiguration(this);
+            config.put("grid", grid.getConfiguration());
+
+            LegendConfiguration legend = new LegendConfiguration(this);
+            config.put("legend", legend.getConfiguration());
+
+            variables.put("type", "'" + this.getType() + "'");
+            variables.put("config", config.toString());
+
+            // For debugging purposes, note that most minified files have un-minified equivalents in the same directory.
+            // To use them, switch affix to "max" and get rid of the ignoreAssetsPattern in build.gradle that
+            // filters them out of the APK.
+            String affix = "min";
+            html.append(
+                    "<html>" +
+                            "<head>" +
+                            "<link rel='stylesheet' type='text/css' href='file:///android_asset/graphing/c3.min.css'></link>" +
+                            "<link rel='stylesheet' type='text/css' href='file:///android_asset/graphing/graph." + affix + ".css'></link>" +
+                            "<script type='text/javascript' src='file:///android_asset/graphing/d3.min.js'></script>" +
+                            "<script type='text/javascript' src='file:///android_asset/graphing/c3." + affix + ".js' charset='utf-8'></script>" +
+                            "<script type='text/javascript'>try {\n");
+
+            html.append(getVariablesHTML(variables, null));
+            html.append(getVariablesHTML(data.getVariables(), "data"));
+            html.append(getVariablesHTML(axis.getVariables(), "axis"));
+            html.append(getVariablesHTML(grid.getVariables(), "grid"));
+            html.append(getVariablesHTML(legend.getVariables(), "legend"));
+
+            String titleHTML = "<div id='chart-title'>" + title + "</div>";
+            String errorHTML = "<div id='error'></div>";
+            String chartHTML = "<div id='chart'></div>";
+            html.append(
+                    "\n} catch (e) { displayError(e); }</script>" +
+                            "<script type='text/javascript' src='file:///android_asset/graphing/graph." + affix + ".js'></script>" +
+                            "</head>" +
+                            "<body>" + titleHTML + errorHTML + chartHTML + "</body>" +
+                            "</html>");
+        } catch (JSONException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
```

Comment:
Not sure; I just copied this code over from one place to another, so this is how it already was, but that seems right to me so I'll change it.